### PR TITLE
Add junit and polarion reporters

### PR DESCRIPTION
**Pull request by qinqon**
It uses the kubevirt's ones

Signed-off-by: Quique Llorente <ellorent@redhat.com>

**What this PR does / why we need it**:

This generate a junit and polarion reports to be consumed by tools and show the results in 
a proper way.

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: automation/check-patch.e2e-okd.sh [hunk 1]
```diff
@@ -22,11 +22,12 @@ main() {
     trap teardown EXIT SIGINT SIGTERM SIGSTOP
     make cluster-sync
     make \
-        E2E_TEST_ARGS='-ginkgo.noColor -ginkgo.skip .*NNS.*cleanup.*|.*OVS.*' \
+        E2E_TEST_ARGS="$E2E_TEST_ARGS -ginkgo.skip .*NNS.*cleanup.*|.*OVS.*" \
         test/e2e
     make \
-        E2E_TEST_ARGS='-ginkgo.noColor -ginkgo.focus .*NNS.*cleanup.*' \
+        E2E_TEST_ARGS="$E2E_TEST_ARGS -ginkgo.focus .*NNS.*cleanup.*" \
```
Comment (phoracek) [inline automation/check-patch.e2e-okd.sh]: With this you overwrite previous junit output, don't you?

File: test/e2e/main_test.go [hunk 2]
```diff
@@ -55,6 +57,16 @@ func TestE2E(tapi *testing.T) {
...
+
+	reporters := []Reporter{}
+	if ginkgo_reporters.Polarion.Run {
+		reporters = append(reporters, &ginkgo_reporters.Polarion)
+	}
+	if ginkgo_reporters.JunitOutput != "" {
+		reporters = append(reporters, ginkgo_reporters.NewJunitReporter())
+	}
+	RunSpecsWithDefaultAndCustomReporters(t, "Tests Suite", reporters)
+
```
Comment (phoracek) [inline test/e2e/main_test.go]: Please drop the line

File: test/e2e/main_test.go [hunk 3]
```diff
@@ -55,6 +57,16 @@ func TestE2E(tapi *testing.T) {
 	t = tapi
 	RegisterFailHandler(Fail)
 	RunSpecs(t, "E2E Test Suite")
+
+	reporters := []Reporter{}
```
Comment (phoracek) [inline test/e2e/main_test.go]: Is not this too late after RunSpecs?

Comment (qinqon) [reply test/e2e/main_test.go]: Ups we are running it twice.

File: automation/check-patch.e2e-okd.sh (hunk 1 above)
Comment (qinqon) [reply automation/check-patch.e2e-okd.sh]: You right, let's just have two different files, we have to follow the pattern at https://github.com/kubevirt/project-infra/blob/master/github/ci/prow/files/config.yaml#L28

File: automation/check-patch.e2e-k8s.sh.orig [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    cd ${TMP_PROJECT_PATH}
+
+    # Let's fail fast if it's not compiling
+    make handler
+
+    make cluster-down
+    make cluster-up
+    trap teardown EXIT SIGINT SIGTERM SIGSTOP
+    make cluster-sync
+<<<<<<< HEAD
```
Comment (phoracek) [inline automation/check-patch.e2e-k8s.sh.orig]: Bad rebase

**Comment by phoracek** [comment]
I don't see the results exported https://gcsweb.apps.ovirt.org/gcs/kubevirt-prow/pr-logs/pull/nmstate_kubernetes-nmstate/310/pull-kubernetes-nmstate-e2e-k8s/1218159961884856320/

File: automation/check-patch.unit-test.sh [hunk 5]
```diff
@@ -1,4 +1,4 @@
-#!/bin/bash -xe
+#!/bin/bash -e
```
Comment (phoracek) [inline automation/check-patch.unit-test.sh]: Why did you drop the x? I find it helpful for debugging.

File: pkg/apis/nmstate/v1alpha1/v1alpha1_suite_test.go [hunk 6]
```diff
@@ -5,9 +5,12 @@ import (
...
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
+
+	"github.com/onsi/ginkgo/reporters"
 )
 
 func TestUnit(t *testing.T) {
 	RegisterFailHandler(Fail)
-	RunSpecs(t, "API Test Suite")
+	junitReporter := reporters.NewJUnitReporter("junit.api.xml")
```
Comment (phoracek) [inline pkg/apis/nmstate/v1alpha1/v1alpha1_suite_test.go]: We should make it so it reflects the full path. Don't make me think of names for junit files, and even worse, decoding which was created by which suite. We can just name this one apis-nmstate-v1alpha1-v1alpha1_suite_test.junit.xml

**Comment by phoracek** [comment]
I don't see any junit.xml collected in https://gcsweb.apps.ovirt.org/gcs/kubevirt-prow/pr-logs/pull/nmstate_kubernetes-nmstate/310/pull-kubernetes-nmstate-e2e-k8s/1219676833675481088/. Also, the won't those two jobs overwrite one another in the artifacts? Or are they kept in separate subdirectories?

File: pkg/apis/nmstate/v1alpha1/v1alpha1_suite_test.go (hunk 6 above)
Comment (qinqon) [reply pkg/apis/nmstate/v1alpha1/v1alpha1_suite_test.go]: Let's just call them junit.xml and copy with full path at artifacts path

**Pull request merged**